geniza/corpus/models.py

```diff
@@ -434,28 +451,78 @@ def editions(self):
+            "tags",
+            "languages",
+            "footnotes",
+            "log_entries",
+            Prefetch(
+                "textblock_set",
+                queryset=TextBlock.objects.select_related(
+                    "fragment", "fragment__collection"
+                ),
+            ),
+        )
 
     def index_data(self):
         """data for indexing in Solr"""
-
         index_data = super().index_data()
+
+        # get fragments via textblocks for correct order
+        # and to take advantage of prefetching
+        fragments = [tb.fragment for tb in self.textblock_set.all()]
         index_data.update(
             {
                 "pgpid_i": self.id,
                 "type_s": self.doctype.name if self.doctype else "Unknown",
                 "description_t": self.description,
-                "notes_t": self.notes,
-                "needs_review_t": self.needs_review,
-                "shelfmark_ss": [f.shelfmark for f in self.fragments.all()],
+                "notes_t": self.notes or None,
+                "needs_review_t": self.needs_review or None,
+                "shelfmark_ss": [f.shelfmark for f in fragments],
+                # library/collection possibly redundant?
+                "collection_ss": [str(f.collection) for f in fragments],
                 "tags_ss": [t.name for t in self.tags.all()],
                 "status_s": self.get_status_display(),
                 "old_pgpids_is": self.old_pgpids,
-                # TODO: editors/translators/sources
             }
         )
 
+        # count scholarship records by type
+        footnotes = self.footnotes.all()
+        counts = defaultdict(int)
+        for fn in footnotes:
+            for val in fn.doc_relation:
+                counts[val] += 1
+        index_data.update(
+            {
+                "num_editions_i": counts[Footnote.EDITION],
+                "num_translations_i": counts[Footnote.TRANSLATION],
+                "num_discussions_i": counts[Footnote.DISCUSSION],
+                "scholarship_count_i": sum(counts.values()),
+                # preliminary scholarship record indexing
+                # (may need splitting out and weighting based on type of scholarship)
+                "scholarship_t": [fn.display() for fn in footnotes],
+            }
+        )
+
+        last_log_entry = self.log_entries.last()
+        if last_log_entry:
+            # TODO: index full date for sorting but only display year
+            index_data["input_year_i"] = last_log_entry.action_time.year
+        # image indexing was causing timeouts with cudl manifests,
+        # hold off on that for now
+        # images = img_labels = []
+        # for fragment in self.fragments.all():
+        #     imgs = fragment.iiif_images()
+        #     if imgs:
+        #         images.extend(imgs[0])
+        #         img_labels.extend(imgs[1])
+        # index_data['images_s'] = images
```

Comment:
what kind of data actually gets indexed in this field?

Reply:
I was thinking to index the IIIF image urls so we could easily generate the thumbnails in the search results without an additional call, but stopped and decided it was out of scope when I got throttled by Cambridge when I was trying to index. I don't think indexing image urls on documents is the right solution, just not sure what is.